### PR TITLE
sql/mysql: add support for charset attributes in sqlspec

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/mysql/sqlspec.go [hunk 1]
```diff
@@ -307,8 +352,10 @@ func binarySpec(t *schema.BinaryType) (*sqlspec.Column, error) {
 func stringSpec(t *schema.StringType) (*sqlspec.Column, error) {
 	switch t.T {
 	case tVarchar, tMediumText, tLongText, tTinyText, tText, tChar:
-		s := strconv.Itoa(t.Size)
-		return specutil.NewCol("", "string", specutil.LitAttr("size", s)), nil
+		if t.Size == 0 {
+			return &sqlspec.Column{Type: t.T}, nil
+		}
+		return specutil.NewCol("", "string", specutil.LitAttr("size", strconv.Itoa(t.Size))), nil
```
Comment (a8m) [inline sql/mysql/sqlspec.go:358]: A tiny patch to support text fields. Will remove after we decide to tackle virtual fields.

File: sql/mysql/sqlspec_test.go [hunk 2]
```diff
@@ -151,6 +151,134 @@ table "accounts" {
...
+    null      = false
+    type      = "text"
+    charset   = "utf8mb4"
+    collation = "utf8mb4_0900_ai_ci"
+  }
+}
+schema "test" {
+  charset   = "utf8mb4"
+  collation = "utf8mb4_0900_ai_ci"
+}
```
Comment (yonidavidson) [inline sql/mysql/sqlspec_test.go:251]: Excellent test

**Review by yonidavidson — APPROVED** [review]
💯 
Not a fan that we need to apply charset on each logical part of the DB (schema, table, column ) but it better be explicit.

**Comment by a8m** [comment]
> 💯 Not a fan that we need to apply charset on each logical part of the DB (schema, table, column ) but it better be explicit.

We don't, it's inherited from the parent element and you can override it for specific elements.

**Pull request merged**